ch_backup/cli.py

```diff
@@ -60,6 +60,28 @@ def signal_handler(signum, _frame):
 signal.signal(signal.SIGINT, signal_handler)
 
 
+class JsonParamType(ParamType):
+    """
+    JsonParamType type for command-line parameter for JSON value.
+    """
+
+    name = "json"
+
+    def convert(self, value, param, ctx):
+        try:
+            if re.fullmatch(
+                r'\s*([\[{"].*|true|false|null|\d+(\.\d+)?)\s*',
+                value,
+                re.MULTILINE | re.DOTALL,
+            ):
+                return json.loads(value)
+            return value.strip()
+        except json.JSONDecodeError as e:
+            raise ClickException(
```

Comment:
The `self.fail()` is [recommended](https://click.palletsprojects.com/en/8.1.x/parameters/#implementing-custom-types) way of signaling the conversion fails.